### PR TITLE
[REG-1626] Enable replay interaction with Coherent GameFace components

### PR DESCRIPTION
**Pull request by abeizer**
Sends text events specifically for GameFace inputs to receive characters and produce `onChange` events. This requires us to take `shift` into consideration so that we know which character to send based on whether `shift` is held at the same time as another key.

https://www.loom.com/share/1084658e11c5412e8d70b6d1fc9abd94

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Editor/Scripts/RGSettingsUIRegistrar.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static SettingsProvider CreateRGSettingsProvider()
             var provider = new SettingsProvider("Regression Games", SettingsScope.Project)
             {
                 // Create the SettingsProvider and initialize its drawing (IMGUI) function in place:
-                guiHandler = async (searchContext) =>
+                guiHandler = (searchContext) =>
```
Comment (abeizer) [inline src/gg.regression.unity.bots/Editor/Scripts/RGSettingsUIRegistrar.cs:28]: `async` without `await` was generating a warning in our BossRoom sample, but the message was an error in a partner's project that prevented play.

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs [hunk 2]
```diff
@@ -200,6 +200,99 @@ public class KeyboardInputActionObserver : MonoBehaviour
             { "OEM4", Key.OEM4 },
             { "OEM5", Key.OEM5 },
         });
+        
+        // Key -> (without-Shift, with-Shift) values
+        public static readonly IReadOnlyDictionary<Key, (char, char)> KeyToValueMap = new ReadOnlyDictionary<Key, (char,char)>(new Dictionary<Key, (char,char)>()
+        {
+            //row 1 (top row)
```
Comment (abeizer) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs:207]: For "standard" keyboards this is the function row. Need to do some testing to see whether we need to handle fn keys.

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs [hunk 3]
```diff
@@ -200,6 +200,99 @@ public class KeyboardInputActionObserver : MonoBehaviour
...
+            { Key.B, ('b','B') },
+            { Key.N, ('n','N') },
+            { Key.M, ('m','M') },
+            { Key.Comma, (',','<') },
+            { Key.Period, ('.','>') },
+            { Key.Slash, ('/','?') },
+            // same for right shift
+            
+            // row 6 (bottom row)
+            // TODO ctrl, alt will be modifier keys??
```
Comment (RG-nAmKcAz) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs:271]: yes, they will be modifier keys

**Comment by vontell** [comment]
@abeizer this has DRAFT in the title, is it ready to review? Also, in the future let's use the built-in github feature of marking it as a draft, which is a little easier to notice as a reviewer.

**Review by vontell — APPROVED** [review]
No additional comments, and just one small thing noted

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs (hunk 2 above)
Comment (abeizer) [reply src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs]: Exactly. We need to select the correct character based on whether `shift` is being held at the same time. I'm sure we could clean up the dictionaries at some point, but we wanted to get this working for our partner project ASAP.

**Comment by abeizer** [comment]
> @abeizer this has DRAFT in the title, is it ready to review? Also, in the future let's use the built-in github feature of marking it as a draft, which is a little easier to notice as a reviewer.

No, I've always used "DRAFT:" to mean "this is 100% a draft". I wasn't aware there was a separate kind of PR. I guess this is a carryover from using GitLab. I must've never opened a draft PR for a repo with codeowners.

**Review by RG-nAmKcAz — APPROVED** [review]

**Review by vontell — APPROVED** [review]

**Pull request merged**